tcclient/creds.go

```diff
@@ -7,16 +7,75 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"net/http"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/taskcluster/jsonschema2go/text"
 	"github.com/taskcluster/slugid-go/slugid"
+	"github.com/tent/hawk-go"
 )
 
-type Credentials struct {
-	ClientId    string `json:"clientId"`
+type Credentials interface {
+	fmt.Stringer
+	ConfigureAuth(httpRequest *http.Request) error
+	getExtField() (string, error)
+	HawkCredentials() *hawk.Credentials
```

Comment:
Oh, we could define an interface:
```go
type PermanentCredentials interface {
  Credentials
  NewTemporaryCredentials(....) (Credentials, error)
}
```

Notice that it implements both interfaces...


----
FYI, adding a `MarshalJSON() ([]byte, error)` method to `Credentials` wouldn't be bad, that way you can always get the credentials out again...

And the above `NewTemporaryCredentials` would work even though it just returns an interface.